core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -312,10 +312,27 @@ trait SubmissionMonitor extends FutureSupport with LazyLogging with RawlsInstrum
   ): Future[Option[WorkflowRecord]] =
     workflowRec.externalId match {
       // fetch cost information for the workflow if submission has a cost cap threshold defined
-      case Some(externalId) if costCapThreshold.isDefined =>
-        executionServiceCluster.getCost(workflowRec, petUser).map { costBreakdown =>
-          Option(workflowRec.copy(status = costBreakdown.status, cost = costBreakdown.cost.some))
-        }
+      case Some(externalId) if perWorkflowCostCap.isDefined =>
+        for {
+          costBreakdown <- executionServiceCluster.getCost(workflowRec, petUser)
+          updatedWorkflowRec <-
+            if (costBreakdown.cost > perWorkflowCostCap.get) {
```

Comment:
nit: you can include `perWorkflowCostCap` in the match then get rid of `if perWorkflowCostCap.isDefined` and this `.get` call